drivers/dma/dma_andes_atcdmacx00.c

```diff
@@ -111,21 +113,10 @@ LOG_MODULE_REGISTER(dma_andes_atcdmac300);
 #define DMA_INT_STATUS_ERROR_VAL(x)	FIELD_GET(DMA_INT_STATUS_ERROR_MASK, (x))
 #define DMA_INT_STATUS_CH_MSK(ch)	(0x111 << ch)
 
-typedef void (*atcdmac300_cfg_func_t)(void);
+typedef void (*atcdmacx00_cfg_func_t)(void);
 
 struct chain_block {
```

Comment:
If this is a linked list transfer descriptor as it seems to be, its unclear why the various struct members are being dropped

Reply:
@teburd  Thank you for your review.
The change is not about moving fields from the `chain_block` struct to `offset_table`, these two structures are for different purposes.
Both changes were made to support multi-serial:
1. The `chain_block` struct was changed to an array because the layout of the chain block differs between ATCDMAC100 and ATCDMAC300. The array size accounts for the maximum required space across both variants.
2. The `offset_table` struct was added to handle differences in register offsets between the 100 and 300 series, which will be set during initialization.

Reply:
I also added the some commentary on these structure.